src/openforms/contrib/auth_oidc/backends.py

```diff
@@ -65,6 +65,45 @@ def get_or_create_user(
         user.is_active = True  # type: ignore
         return user
 
+    def _get_form_authentication_backend(
+        self, plugin_identifier: str
+    ) -> FormAuthenticationBackend | None:
+        return_url = self.request.session.get(_RETURN_URL_SESSION_KEY, "")
+        return_path = furl(return_url).path
+        _, _, kwargs = resolve(return_path)
+
+        try:
+            form = Form.objects.get(slug=kwargs.get("slug"))
+
+            auth_backend = FormAuthenticationBackend.objects.get(
+                form=form, backend=plugin_identifier
+            )
+        except (Form.DoesNotExist, FormAuthenticationBackend.DoesNotExist):
+            return None
+
+        return auth_backend
+
+    def _obfuscate_claims(self, claims: JSONObject):
+        # Get plugin
+        config_to_plugin = get_config_to_plugin()
+        assert self.config_class and self.config_class in config_to_plugin
+        plugin = config_to_plugin[self.config_class]
```

Comment:
if this is repeated multiple times, you might want to consider putting it in a property:

```py
@property
def auth_plugin(self):
    config_to_plugin = get_config_to_plugin()
    assert self.config_class and self.config_class in config_to_plugin
    return config_to_plugin[self.config_class]
```

this has the potential to be upgraded to `@cached_property` in the future if it's an expensive call

You can then replace your code with:


```suggestion
        plugin = self.auth_plugin
```